batch/batch/google_storage.py

```diff
@@ -95,10 +132,32 @@ def _delete_gs_files(self, uri_prefix):
                 continue
 
     def _delete_gs_file(self, uri):
-        bucket, path = GCS._parse_uri(uri)
-        bucket = self.gcs_client.bucket(bucket)
-        f = bucket.blob(path)
+        f = self._get_blob(uri)
         try:
             f.delete()
         except google.api_core.exceptions.NotFound:
             return
+
+    def _copy_gs_file(self, src, dest, *args, **kwargs):
+        src_bucket, src_path = GCS._parse_uri(src)
+        src_bucket = self.gcs_client.bucket(src_bucket)
+        dest_bucket, dest_path = GCS._parse_uri(dest)
+        dest_bucket = self.gcs_client.bucket(dest_bucket)
+        src_f = src_bucket.blob(src_path)
+        src_bucket.copy_blob(src_f, dest_bucket, new_name=dest_path, *args, **kwargs)
+
+    def _list_gs_files(self, uri, max_results=None):
```

Comment:
this is `_recursively_list_all_gs_files` or `_list_tree_gs_file` or something like that. I think we should be careful with terminology here and use `list` only for the equivalent of `ls` or `listdir`